oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/model/StorageModels.java

```diff
@@ -250,6 +250,14 @@ private void retrieval(final Class<?> clazz,
             }
         }
 
+        if (clazz.isAnnotationPresent(SQLDatabase.Sharding.class)) {
+            SQLDatabase.Sharding sharding = clazz.getAnnotation(SQLDatabase.Sharding.class);
```

Comment:
`SQLDatabase.Sharding.class` is present in the parent class. `FunctionCategory#doGetAnnotation` use a recursion to read **MeterFunction** annotation from parent class.